### DISCUSSION TITLE
x-on Shorthand Not Work inside PJAX Loaded Page

### DISCUSSION THREAD

**Discussion by yllumi** [alpinejs/alpine#3669] [7. Bugs (archived)] [answered]
I am currently working on a project using PHP for server-side rendering and PJAX for loading page shells. The page content is handled with the help of Alpinejs. There is a particular condition where, upon refreshing the page, all DOM events can be executed using shorthand notations like @click and @keydown.enter. However, when the page is loaded via AJAX using PJAX, all DOM events using shorthand notations do not work. Nevertheless, when I change all shorthand @ to x-on:, everything functions as usual.

I don't know if this is a bug or if I always have to use x-on: in situations like this. Because the shorthand for x-bind works perfectly fine on the AJAX-loaded page.

**Comment by ekwoka** [comment, marked as answer]
I'd expect that the @click whatever is being removed by pjax or whatever.

Did you check the actual dom?

**Reply by yllumi** [reply]
Wow, you were right! The @click attribute disappeared on next load by pjax
Thanks for the explanation
